ui/pages/permissions-connect/flask/snap-install/snap-install.js

```diff
@@ -0,0 +1,121 @@
+import PropTypes from 'prop-types';
+import React, { useCallback, useMemo, useState } from 'react';
+import { PageContainerFooter } from '../../../../components/ui/page-container';
+import PermissionsConnectPermissionList from '../../../../components/app/permissions-connect-permission-list';
+import PermissionsConnectFooter from '../../../../components/app/permissions-connect-footer';
+import PermissionConnectHeader from '../../../../components/app/permissions-connect-header';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+import SnapInstallWarning from '../../../../components/app/flask/snap-install-warning';
+
+export default function SnapInstall({
+  request,
+  approveSnapInstall,
+  rejectSnapInstall,
+  targetSubjectMetadata,
+}) {
+  const t = useI18nContext();
+
+  const [isShowingWarning, setIsShowingWarning] = useState(false);
+
+  const onCancel = useCallback(() => rejectSnapInstall(request.metadata.id), [
+    request,
+    rejectSnapInstall,
+  ]);
+
+  const onSubmit = useCallback(() => approveSnapInstall(request), [
+    request,
+    approveSnapInstall,
+  ]);
+
+  const npmId = useMemo(() => {
+    if (!targetSubjectMetadata.origin.startsWith('npm:')) {
+      return undefined;
+    }
+    return targetSubjectMetadata.origin.substring(4);
+  }, [targetSubjectMetadata]);
+
+  const shouldShowWarning = useMemo(
+    () =>
+      Boolean(
+        request.permissions &&
+          Object.keys(request.permissions).find((v) =>
+            v.startsWith('snap_getBip44Entropy_'),
+          ),
+      ),
+    [request.permissions],
+  );
+
+  return (
+    <div className="page-container snap-install">
+      <div className="headers">
+        <PermissionConnectHeader
+          icon={targetSubjectMetadata.iconUrl}
+          iconName={targetSubjectMetadata.name}
+          headerTitle={t('snapInstall')}
+          headerText={null} // TODO(ritave): Add header text when snaps support description
+          siteOrigin={targetSubjectMetadata.origin}
+          npmPackageName={npmId}
+        />
+        <div className="snap-requests-permission">
+          {t('snapRequestsPermission')}
+        </div>
```

Comment:
[REQ] Use Typography component and its props to achieve styles and use an appropriate tag for text contents.